### ISSUE TITLE
Lua WheelBase:GetWheelItem( int iIndex ) problems

### ISSUE THREAD

**Issue by SpoOkyMagician** [stepmania/stepmania#520]
I did some experimenting with the GetWheelItem and I got really confused trying to get to WheelItemBase:IsLoaded...

I did some trace stuff and I found out some interesting things... Wheel Items do actually have a index but if you try to use that specific index, it warns that you are out of bounds. I was curious to see if I could find ANYTHING that would work. Turns out 0 does. (Managed to get WheelItemBase finally.)

What is the point in having an index if only 0 works? Is this intentional for SM5? If so, you might want to change the index thing a bit then. If it's not, you got a really annoying bug on your hands...

**Comment by dguzek** [comment]
See: [WheelBase.cpp lines 101-106](https://github.com/stepmania/stepmania/blob/master/src/WheelBase.cpp#L101-L106)

**Comment by kyzentun** [comment]
This sounds related to #385.  I think the general work around is to avoid trying to grab the music wheel and its items during a CurrentSongChangeMessageCommand because the wheel changes the current song while it is loading, before its safe to use.

dguzek's comment explains why your transform command isn't actually passed the item index or the number of items (it's a really dumb excuse, and a large part of why I don't use the music wheel).

In general, nothing I want to do to the music wheel can be done without breaking compatibility, so it's going to keep its strange problems until SM6 or later.

**Comment by SpoOkyMagician** [comment]
...Really? *shrugs* The only reason why I am trying to get this is because I am still having issues with my radar values. (They are not being set correctly in my theme upon changing the song/difficulty.) I thought this would help or something... I guess I was wrong... I tried some sleep commands as well and that didn't help much either. I'll have to experiment some more I guess...

**Comment by kyzentun** [comment]
If you're fetching radar values, you shouldn't need to look at the music wheel for anything.
Just use GAMESTATE:GetCurrentSteps(player_number), then fetch the radar values from the Steps that returns.  If GetCurrentSteps returns nil, then set your radar display thing to empty.

**Comment by SpoOkyMagician** [comment]
No, it's a custom radar value thing. (Besides, I already did/checked that.)

**Comment by kyzentun** [comment]
I figured it was a custom radar thing, because frankly, writing a custom one is better than using the built in groove radar actor.

Here's what I do:
```
local steps= GAMESTATE:GetCurrentSteps(player_number)
if steps then
	local radars= steps:GetRadarValus(player_number)
	local stream= radars:GetValue("RadarCategory_Stream")
	-- and use stream to set text number.
end
```

**Comment by SpoOkyMagician** [comment]
I can get the number and whatnot with no problem. It's just when I change song/difficulty. They grab the wrong values. (As I said, I tried sleep and it helps some but, they still seem to be wrong at times.)
